### ISSUE TITLE
plugin-compatibility-matrix: Linux

### ISSUE THREAD

**Issue by dromer** [izzyreal/vmpc-juce#11]
So, just a quick test in jack on linux: with the native VST2 in Carla I see 2 channels input and 10 channels output.

When I play a sample (ie. a 16bit 44.1khz file from disk) it is played over the stereo outputs. Assigning it to a pad it can either be panned over the stereo outputs, or to any of the 8 single channel output expansions (or both at the same time).

Seems to work great.

In the compat matrix it's maybe better to call the outputs 2x stereo + 8x mono ? (as that is the implementation in the mpc).

**Comment by dromer** [comment]
The standalone application is not working so well however.

In alsa I think that only stereo in/out works, the sound is terribly crackly though.
If I try to connect the output in jack the standalone application crashes.

**Comment by izzyreal** [comment]
Reply to VST2/Carla: Nice it's working that well. As to the outputs, you mean 2x stereo because MAIN + HEADPHONES? 8x MONO is a possibility, but so far I've approached it from a stereo pair perspective. I may have to revisit that.

Reply to standalone: I haven't tried it extensively in Linux yet. So I still don't know with what settings/drivers it works best. I suspect best results with Jack. If you're hearing crackles with Alsa, does it help to increase the buffer size?

For the crash when trying to connect the output in jack I've raised an issue here: https://github.com/izzyreal/mpc/issues/62

**Comment by dromer** [comment]
Oh, no I meant 1x stereo in and 1x stereo out. and 8x mono output.
That's how the mpc designates the routings in the mixer no?

I get either sound (with panning controls) over the stereo output (channels 1 and 2) or I can route to the external channels 1-8.

[edit: actually by default it routs to the extra outputs in pairs, but only with a mono sample it automatically pics a single output channel .. I think. Is this correct?]

**Comment by izzyreal** [comment]
Yes, if you have a stereo sample assigned to a pad, that pad's mixer channel will assign to individual outputs in pairs 1/2, 3/4, 5/6 and 7/8. For mono it's 1 - 8 individually. So it does both. Maybe I can offer mono channels that are duplicates of the stereo ones or vice versa.

**Comment by dromer** [comment]
Well in jack I just see 10 output channels, of which the first 2 correspond to the Stereo Main.
I just didn't realize that for stereo samples the outputs of the expansion board are automatically combined :)

(I actually have an IB208P in my S2000, but I never used it with stereo samples. don't know if the S2000 does the same)

**Comment by izzyreal** [comment]
The expansion board in my MPC2000XL comes from an S2000 :) I would not be surprised if it does the same.

**Comment by izzyreal** [comment]
Pretty old ticket, sorry for taking so long and also I'm the messenger of bad news: https://github.com/izzyreal/vmpc-juce/blob/master/plugin-compatibility-matrix.md

tl;dr The only plugin/host combo that works well (for me on Arch at least) is LV2 in Carla.

With regard to the standalone version, I think this one is pretty stable now. Let me know if that's not the case for you.

**Comment by dromer** [comment]
There are many other LV2 hosts, though. At least Reaper, Mixxx and soon Bespoke Synth (JUCE7 base).

BTW if/when you are moving to JUCE7 it would be nice if you enable LV2 for all supported platforms.

**Comment by izzyreal** [comment]
Reaper, Bitwig, Qtractor and Mixxx empty stubs are added to the matrix. Not sure when I'm going to test and fill them in though.

I agree it would be nice to have LV2 for all platforms that support LV2, but I think at the moment LV2 in JUCE is only supported on Linux. I could be wrong though.

**Comment by dromer** [comment]
Very wrong, it's supported for Windows, MacOS and Linux. :)

**Comment by izzyreal** [comment]
> Very wrong, it's supported for Windows, MacOS and Linux. :)

On Windows I get an LV2 target in the build definition, but when trying to build it I hit https://github.com/juce-framework/JUCE/blob/4c43bf429e90690cb1f05b7c8a044cc9f5a59e7d/modules/juce_core/native/juce_BasicNativeHeaders.h#L123

On macOS I don't get an LV2 target in the build definition.

So if JUCE supports LV2 on those platforms it's not clear to me how to make use of it.

> Bitwig does not support LV2, btw.

Thanks, I have amended the compatibility matrix.
